networks:
    ai_network:
        driver: bridge

services:
    facefusion:
        image: facefusion:3.1.1
        container_name: facefusion
        restart: always
        runtime: nvidia
        privileged: true
        volumes:
            - d:/AI/facefusion3/results:/code/results
            - d:/AI/facefusion3/error:/code/error
            - d:/AI/facefusion3/logs:/code/logs
        environment:
            - TQDM_DISABLE=1 # 全局关闭进度条显示
            - PYTHONUNBUFFERED=1 # 取消标准输出缓冲
            - CUDA_VISIBLE_DEVICES=1
            - PYTORCH_CUDA_ALLOC_CONF=max_split_size_mb:1024
        tty: true  # 分配伪终端
        deploy:
            resources:
                reservations:
                    devices:
                        -   driver: nvidia
                            capabilities: [ gpu ]
        shm_size: '32g'
        ports:
            - '7864:7864'
        command: python /code/api.py
        networks:
            - ai_network
